### PR TITLE
Mac Dev PR 1

### PR DESCRIPTION
**Pull request by archbirdplus**
Mac support should no longer be a Cubyz issue with this PR.

Contributes to #324, #325, #299, but doesn't resolve them.

Note: I use a relative path in my new `deps_local`, which looks like it might break on other systems.

### CODE REVIEW

File: build.zig.zon [hunk 1]
```diff
@@ -3,10 +3,14 @@
 	.version = "0.0.0",
 	.paths = .{""},
 	.dependencies = .{
+		.deps_local = .{
+			.url = "file:../Cubyz-libs/zig-out.tar.gz", // macos has no official deps release
+			.hash = "12209f7db8911ff57c69db2097437ad5e00141e862b0e760926c966b8652d09edf43",
+		},
 		.deps = .{
-			//.url = "file:///home/mint/Cubyz-libs/zig-out.tar.gz", // fetching the lib from the local folder
+			// url = "file:../Cubyz-libs/zig-out.tar.gz", // fetching the lib from the local folder
```
Comment (IntegratedQuantum) [inline build.zig.zon:11]: Could you please keep the dot in front of the `url` field? Makes it easier to swap the two cases.

Comment (archbirdplus) [reply build.zig.zon]: Updated. But does it work?

Comment (IntegratedQuantum) [reply build.zig.zon]: Yeah, it appears to work.

File: build.zig.zon [hunk 2]
```diff
@@ -3,10 +3,14 @@
 	.version = "0.0.0",
 	.paths = .{""},
 	.dependencies = .{
+		.deps_local = .{
+			.url = "file:../Cubyz-libs/zig-out.tar.gz", // macos has no official deps release
```
Comment (IntegratedQuantum) [inline build.zig.zon:7]: This does not work though. It seems zig fetches the packages independently of the target, giving me an error if the path doesn't exist.

Comment (archbirdplus) [reply build.zig.zon]: I thought it was lazy by default. Added laziness now.

**Comment by IntegratedQuantum** [comment]
It works now. Thank you!

**Pull request merged**